test/e2e/replication/replication_test.go

```diff
@@ -0,0 +1,1469 @@
+
+		replicaSet := &appsv1.ReplicaSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: replicaSetName,
+				Labels: map[string]string{
+					"app": "multi-container-test",
+				},
+			},
+			Spec: appsv1.ReplicaSetSpec{
+				Replicas: &replicaCount,
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"app": "multi-container-test",
+					},
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"app": "multi-container-test",
+						},
+					},
+					Spec: v1.PodSpec{
+						InitContainers: []v1.Container{
+							{
+								Name:    "init-container",
+								Image:   testImage,
+								Command: []string{"sh", "-c", "echo 'Init complete' && sleep 2"}, // Reduced sleep
+							},
+						},
+						Containers: []v1.Container{
+							{
+								Name:    "main-container",
+								Image:   testImage,
+								Command: []string{"sleep", "300"},
+								Resources: v1.ResourceRequirements{
+									Requests: v1.ResourceList{
+										v1.ResourceCPU:    resource.MustParse("25m"),  // Reduced from 50m
+										v1.ResourceMemory: resource.MustParse("32Mi"), // Reduced from 64Mi
+									},
+								},
+							},
+							{
+								Name:    "sidecar-container",
+								Image:   testImage,
+								Command: []string{"sh", "-c", "while true; do echo 'Sidecar running'; sleep 30; done"},
+								Resources: v1.ResourceRequirements{
+									Requests: v1.ResourceList{
+										v1.ResourceCPU:    resource.MustParse("10m"),  // Reduced from 25m
+										v1.ResourceMemory: resource.MustParse("16Mi"), // Reduced from 32Mi
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		ginkgo.By("Creating the multi-container ReplicaSet")
+		_, err := kClient.CreateReplicaSet(replicaSet, dev)
+		gomega.Ω(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Waiting for all multi-container pods to be created and running")
+		var podList *v1.PodList
+		gomega.Eventually(func() bool {
+			podList, err = kClient.ListPods(dev, "app=multi-container-test")
+			if err != nil {
+				return false
+			}
+			if len(podList.Items) != int(replicaCount) {
+				return false
+			}
+			// Check all pods are running
+			for _, pod := range podList.Items {
+				if pod.Status.Phase != v1.PodRunning {
+					return false
+				}
+				// Check init container completion
+				if len(pod.Status.InitContainerStatuses) > 0 {
+					initStatus := pod.Status.InitContainerStatuses[0]
+					if initStatus.State.Terminated == nil || initStatus.State.Terminated.ExitCode != 0 {
+						return false
+					}
+				}
+				// Check main containers
+				if len(pod.Status.ContainerStatuses) != 2 {
+					return false
+				}
+				for _, containerStatus := range pod.Status.ContainerStatuses {
+					if !containerStatus.Ready {
+						return false
+					}
+				}
+			}
+			return true
+		}, longTimeout, pollInterval).Should(gomega.BeTrue(), "All multi-container pods should be ready")
```

Comment:
Ok, this probably needs a different helper function than the previous ones. Regardless, this logic should not be here, but inside `KubeCtl`.